### DISCUSSION TITLE
Theoretical Explanations

### DISCUSSION THREAD

**Discussion by shahjesal15** [Dlloydev/sTune#15] [Q&A]
Hey, your library is absolutely beautiful but i would love to understand it thoroughly can you please collect and provide all the resources you have to built this library it'll be a great help or if you have any blog or something where you or someone explains applied control systems.
Thank you in advance.

**Comment by Dlloydev** [comment]
Thank you for your interest. I've developed the library to see if an automated open loop method of determining the parameters of a FOPDT system was feasible. One of the resources I've looked through is [here](https://apmonitor.com/pdc/index.php/Main/FirstOrderGraphical) as I have the TClab hardware.

The main objective for sTune was to determine all parameters at the inflection point rather than performing a full response test. The only parameter the user needs to know before running an inflection point test is the process time constant. This can be an estimate, but can more accurately be determined by running a full test once, then use the reported time constant in the setup for all future IP tests (which are much faster).

I just noticed that the first reference link in the readme is broken, here's the fixed link:
[Comparison of PID Controller Tuning Methods](http://maulana.lecture.ub.ac.id/files/2014/12/Jurnal-PID_Tunning_Comparison.pdf)

My background isn't in applied control systems, so I'm not really qualified to explain any theoretical details, so you'll need to look through the references provided.

Later this year I should get a chance to update the documentation and examples.

Thanks again,
David Lloyd
